python_modules/libraries/dagster-dask/dagster_dask/data_frame.py

```diff
@@ -29,852 +33,308 @@
-                            and the number of cores up to a maximum of 64MB.
-                            Can be a number like 64000000` or a string like ``'64MB'.
-                            If None, a single block is used for each file.
-                        """,
-                    ),
-                    "sample": Field(
-                        Int,
-                        is_required=False,
-                        description="Number of bytes to use when determining dtypes.",
-                    ),
-                    "assume_missing": Field(
-                        Bool,
-                        is_required=False,
-                        description="""
-                            If True, all integer columns that aren’t specified in dtype are assumed
-                            to contain missing values, and are converted to floats.
-                            Default is False.
-                        """,
-                    ),
-                    "storage_options": Field(
-                        Permissive(),
-                        is_required=False,
-                        description="""
-                            Extra options that make sense for a particular storage connection,
-                            e.g. host, port, username, password, etc.
-                        """,
-                    ),
-                    "include_path_column": Field(
-                        Any,
-                        is_required=False,
-                        description="""
-                            bool or str, Whether or not to include the path to each particular file.
-                            If True a new column is added to the dataframe called path.
-                            If str, sets new column name. Default is False.
-                        """,
-                    ),
-                }
-            ),
-            "orc": Permissive(
-                {
-                    "path": Field(
-                        Any,
-                        is_required=True,
-                        description="""
-                            str or list, Location of file(s),
-                            which can be a full URL with protocol specifier,
-                            and may include glob character if a single string.
-                        """,
-                    ),
-                    "columns": Field(
-                        list, is_required=False, description="Columns to load. If None, loads all.",
-                    ),
-                    "storage_options": Field(
-                        Permissive(),
-                        is_required=False,
-                        description="Further parameters to pass to the bytes backend.",
-                    ),
-                }
-            ),
+    return Shape({
+        "read": Field(
+            Selector(read_fields),
+            is_required=False,
+        ),
+        **{
```

Comment:
should this be `Selector` ? we only expect one to be set here right?

Reply:
Ah, it's a `Shape` because there were additional options that have been held back for later PRs. I'll change it to a `Selector` here, and then change it to `Shape` in the next PR!